### PR TITLE
Release 0.8.2

### PR DESCRIPTION
**Pull request by blakeblackshear**
Features:
- Lots of frontend improvements
- Switches in WebUI for detection/clips/snapshots
- Proactive cache cleanup when >90% full
- Attempt to cleanup overlapping events from the database
- Add disk usage stats
- Global object masks
- Try to avoid detection of stationary objects
- `required_zones` setting for snapshots/clips/mqtt

Changes:
- `max_disappeared` is now 5x FPS by default
- Implement mqtt via websockets for frontend use
- Reduce jpg image sizes to reduce bandwidth
- Unit tests and automated tests for webUI

Fixes:
- Resubscribe to MQTT topics on reconnect

### CODE REVIEW

**Comment by mitchross** [comment]
When do you expect this change to merge and release? Just curious.

**Comment by blakeblackshear** [comment]
Within a weekish. No promises.

**Comment by blakeblackshear** [comment]
@paularmstrong we need the websocket connection to dynamically choose between `ws` and `wss`. I run the webui behind a reverse proxy with tls, and i get the following error:
```
The page at 'https://-----/' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://------/ws'. This request has been blocked; this endpoint must be available over WSS
```

File: web/src/api/mqtt.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function MqttProvider({
   config,
   children,
   createWebsocket = defaultCreateWebsocket,
-  mqttUrl = `${baseUrl.replace(/^https?:/, 'ws:')}/ws`,
+  mqttUrl = `${baseUrl.replace(/^http/, 'ws')}/ws`,
```
Comment (blakeblackshear) [inline web/src/api/mqtt.jsx:37]: @paularmstrong this works. Let me know if you see a problem with this.

Comment (paularmstrong) [reply web/src/api/mqtt.jsx:37]: Nope, no problem!

This should also be better for the HA addon. I haven't been able to test that yet. Do you want to publish this as a beta first?

**Pull request merged**